### PR TITLE
feat: Improve strong-typing of Schema types

### PR DESCRIPTION
**Pull request by ffMathy**
- [X] I have added automatic tests where applicable
- [X] The PR title is suitable as a release note
- [X] The PR contains a description of what has been changed
- [X] The description contains manual test instructions

## Changes

I am improving the typings of mappings as specified in https://help.ftrack.com/en/articles/1040505-schemas.

## Test

Check that code still compiles, and tests run. This has already been done.

### CODE REVIEW

File: source/types.ts [hunk 1]
```diff
@@ -161,8 +161,27 @@ export interface MutationOptions {
   decodeDatesAsIso?: boolean;
 }
 
+type SimpleTypeSchemaProperty = {
+  type: "string" | "boolean" | "number" | "array";
```
Comment (gismya) [inline source/types.ts:165]: There are a few more types available that are not represented on the support page you were referencing, also "array" accidentally snuck in there:
```suggestion
  type: "string" | "boolean" | "number" | "integer" | "variable";
```

**Review by gismya — CHANGES_REQUESTED** [review]
Thank you very much!
Looks great, just a few minor details to adjust.

File: source/types.ts (hunk 1 above)
Comment (ffMathy) [reply source/types.ts]: Thanks! Silly me - didn't see that. Nicely spotted!

**Comment by ffMathy** [comment]
@gismya thanks a bunch for your feedback! I applied the suggested corrections now.

**Review by gismya — APPROVED** [review]
Looks good. Great initiative!

**Comment by gismya** [comment]
I'm having a colleague do a second round, and if I didn't miss anything obvious he'll merge it. Then we'll trigger a release, which should all be done before the end of the day.

**Review by jimmycallin — APPROVED** [review]
looking good, thanks for your contribution! i'll merge this and make a new release :)

**Pull request merged**